### PR TITLE
CRv2: Process hoc_organizer_years attribute

### PR DESCRIPTION
**Pull request by hacodeorg**
[PLC-918](https://codedotorg.atlassian.net/browse/PLC-918): Use contact's `hoc_organizer_years` to set Pardot prospect's `db_Hour_of_Code_Organizer`. 
* Contact's `hoc_organizer_years` value is deducted from `pegasus.forms#kind`.
* Prospect's [db_Hour_of_Code_Organizer](https://pi.pardot.com/prospectFieldCustom/read/id/6731) is a multi-value/multi-select attribute.

## Links
[Speadsheet with all attributes to process](https://docs.google.com/spreadsheets/d/10oKPBVlC5LZee9WiECwd745sX-4EXFALuSIVVwdW5tc/edit#gid=0).

## Test story
Test on an adhoc with connection to a production-cloned db
  ```ruby
  @limit = 100
  cr = ContactRollupsV2.new(is_dry_run: true, limit_extraction: @limit);

  # clean up existing tables
  cr.truncate_or_delete_table ContactRollupsRaw
  cr.truncate_or_delete_table ContactRollupsProcessed

  # extract only data used by this PR
  ContactRollupsRaw.extract_pegasus_forms(@limit)

  # dry run. Verify the sample queries sent to Pardot
  cr.process_contacts
  cr.sync_updated_contacts_with_pardot
  cr.sync_new_contacts_with_pardot
  cr.report_results

  # Verify content in ContactRollupsProcessed, ContactRollupsPardotMemory
  ```

### CODE REVIEW

**Comment by bencodeorg** [comment]
Love how clean the manual testing process is now :)

**Comment by bencodeorg** [comment]
Does putting limit = 100 on extraction queries just give you the first results (ie, the oldest)? Wonder how we could get a more random sample set, quick google suggests most solutions aren't trivial / might be computationally expensive.

File: dashboard/test/models/contact_rollups_processed_test.rb [hunk 1]
```diff
@@ -185,6 +187,28 @@ class ContactRollupsProcessedTest < ActiveSupport::TestCase
...
+          {'value' => 'CSEdWeekEvent2013'},
+          {'value' => 'CSEdWeekEvent2013'},
+          # invalid values
+          {'value' => nil},
+          {'value' => 'HocSignup'},
+          {'value' => 'HocCensus2017'}
+        ]
+      }
+    }
+    expected_output = {hoc_organizer_years: '2013,2019'}
```
Comment (bencodeorg) [inline dashboard/test/models/contact_rollups_processed_test.rb:206]: We don't have any multiselect fields where the field would have a comma in them, do we?

**Review by bencodeorg — APPROVED** [review]

File: dashboard/test/models/contact_rollups_processed_test.rb (hunk 1 above)
Comment (hacodeorg) [reply dashboard/test/models/contact_rollups_processed_test.rb:206]: We don't. There are only 4 multi-select fields (pd enrolled, pd attendend, HoC years, and roles) and we control the values put in them.

**Comment by hacodeorg** [comment]
> Does putting limit = 100 on extraction queries just give you the first results (ie, the oldest)? Wonder how we could get a more random sample set, quick google suggests most solutions aren't trivial / might be computationally expensive.

I think so, it will just give you the first 100. We can sample rows using row id, for example only select rows with `id % 10 == 1`.

**Pull request merged**